[ci] remove IN_CI env var

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #79294
* #79289
* __->__ #79229
* #79290
* #79247

The conventional env var to set is CI. Both circle and GHA set it, so
IN_CI is unnecessary

suo committed:
[ci] remove IN_CI env var

The conventional env var to set is CI. Both circle and GHA set it, so
IN_CI is unnecessary

[ghstack-poisoned]

suo committed:
Update on "[ci] remove IN_CI env var"


The conventional env var to set is CI. Both circle and GHA set it, so
IN_CI is unnecessary

[ghstack-poisoned]

suo committed:
Update on "[ci] remove IN_CI env var"


The conventional env var to set is CI. Both circle and GHA set it, so
IN_CI is unnecessary

[ghstack-poisoned]

suo committed:
Update on "[ci] remove IN_CI env var"


The conventional env var to set is CI. Both circle and GHA set it, so
IN_CI is unnecessary

[ghstack-poisoned]

suo committed:
Update on "[ci] remove IN_CI env var"


The conventional env var to set is CI. Both circle and GHA set it, so
IN_CI is unnecessary

[ghstack-poisoned]

janeyx99 commented on .github/actions/setup-linux/action.yml:
```diff
@@ -45,3 +45,4 @@ runs:
       shell: bash
       run: |
         env | grep '^GITHUB' > "/tmp/github_env_${GITHUB_RUN_ID}"
+        env | grep '^CI' > "/tmp/github_env_${GITHUB_RUN_ID}"
```
Confirming my understanding: this is where the variable `CI` is set, and the goal of this PR is to replace all IN_CI refs with CI and remove the explicit passing we do into the Docker images.

suo replied:
No, the `CI` variable is set by the CI provider: 
- [Github](https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables)
- [Circle](https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables)

Here, the CI variable is written to an `--env-file` that we pass to docker, which I think we should do in general instead of passing env vars through manually.

janeyx99 commented on torch/testing/_internal/common_utils.py:
```diff
@@ -79,8 +79,7 @@
 
-# Environment variable `IN_CI` is set in `.jenkins/common.sh`.
-IS_IN_CI = os.getenv('IN_CI') == '1'
+IS_CI = os.getenv('CI') == '1'
```
Where do we actually set CI to be 1? 

Looking at the test logs, every file should be run with --import-disabled-tests and --import-slow-tests but that's not happening.

suo replied:
Can you elaborate? What are you expecting in the logs vs. what you're seeing?

suo commented:
Hmmm, it does look like the `CI` variable is not getting sent to docker somehow. I will look further.

suo committed:
Update on "[ci] remove IN_CI env var"


The conventional env var to set is CI. Both circle and GHA set it, so
IN_CI is unnecessary

[ghstack-poisoned]

suo commented:
Ah, it's because GITHUB sets `CI` to `true`, but we are checking for it to be `1`.

suo committed:
Update on "[ci] remove IN_CI env var"


The conventional env var to set is CI. Both circle and GHA set it, so
IN_CI is unnecessary

[ghstack-poisoned]

suo committed:
Update on "[ci] remove IN_CI env var"


The conventional env var to set is CI. Both circle and GHA set it, so
IN_CI is unnecessary

[ghstack-poisoned]

suo committed:
Update on "[ci] remove IN_CI env var"


The conventional env var to set is CI. Both circle and GHA set it, so
IN_CI is unnecessary

[ghstack-poisoned]

suo committed:
Update on "[ci] remove IN_CI env var"


The conventional env var to set is CI. Both circle and GHA set it, so
IN_CI is unnecessary

[ghstack-poisoned]

suo committed:
Update on "[ci] remove IN_CI env var"


The conventional env var to set is CI. Both circle and GHA set it, so
IN_CI is unnecessary

[ghstack-poisoned]

suo committed:
Update on "[ci] remove IN_CI env var"


The conventional env var to set is CI. Both circle and GHA set it, so
IN_CI is unnecessary

[ghstack-poisoned]

suo commented:
@pytorchbot merge -g

suo commented:
@pytorchbot rebase -s

suo committed:
Update on "[ci] remove IN_CI env var"


The conventional env var to set is CI. Both circle and GHA set it, so
IN_CI is unnecessary

[ghstack-poisoned]

suo commented:
@pytorchbot merge
